src/buckets/actions/thunks.ts

```diff
@@ -112,6 +115,8 @@ export const createBucket = (bucket: OwnBucket) => async (
 
     dispatch(addBucket(newBucket))
     dispatch(checkBucketLimits())
+
+    callback && callback(resp.data as Bucket)
```

Comment:
how come a callback and not stringing together a bunch of awaits like:
```
await dispatch(addBucket(newBucket))
await dispatch(checkBucketLimits())
return newBucket
```

you also most def want to use the normalized bucket and not resp.data

Reply:
@121watts suggested not returning a bucket from the thunk, so I opted for a generic callback approach.

At first I was using the normalized bucket but every time I tried to extract the `bucket` property from it, it returned `undefined` so I gave up on that

Reply:
what was the reasoning behind not returning the bucket?

you should totally revisit the normalized form.

 after chasing around a lot of "sometimes this returns null instead of a string" bugs, we dropped that layer in to decouple the frontend from the api, and lean pretty heavy on it. normalizing also creates a guarantee that, if you start passing objects around in the view layer, the next component that tries to reference the state will have the same data. imagine: you take resp.data, and it has a property `name` that you use. a month later, you see that code in the view layer, and try to make a selector that looks up the bucket by name in the redux resources in order to grab it's id. except the normalized form doesn't use `name` anymore, it uses `display` for some reason. now it's all broken because we've created two definitions of what a bucket is in two places.
I wish this was just an esoteric thought experiment, but creating multiple sources of truth is at the core of a lot of our reusability / defect issues within the current system